src/tools/FreehandTool.js

```diff
@@ -612,17 +615,40 @@ export default class FreehandMouseTool extends BaseAnnotationTool {
     data.handles.invalidHandlePlacement = freehandIntersect.modify(data.handles, currentHandle);
     data.active = true;
     data.highlight = true;
-    data.handles[currentHandle].x = config.mouseLocation.handles.start.x;
-    data.handles[currentHandle].y = config.mouseLocation.handles.start.y;
-
-    if (currentHandle !== 0) {
-      const lastLineIndex = data.handles[currentHandle - 1].lines.length - 1;
-      const lastLine = data.handles[currentHandle - 1].lines[lastLineIndex];
+    
+    if(eventData.type == EVENTS.TOUCH_DRAG){
```

Comment:
When you bind `_editMouseDragCallback` to specific events, you can set an parameter to indicate if the event is touch or mouse instead of checking for this here? Similarly, if you feel the method deviates enough, you can separate out the touch/mouse logic into separate methods that are called after the shared logic is executed? (This might make unit testing a bit easier, and the code easier to read)

Reply:
ok, methods separated